### PR TITLE
[Numpy] Implementation npx.{sample}_n

### PR DESCRIPTION
**Pull request by xidulu**
## Description ##
See #16793 for description.

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at https://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: tests/python/unittest/test_numpy_op.py [hunk 1]
```diff
@@ -2669,6 +2669,45 @@ def hybrid_forward(self, F, loc, scale):
...
+    
+    for bshape, eshape, dtype, op in itertools.product(batch_shapes, event_shapes, dtypes, op_names):
+        for hybridize in [True, False]:
+            net = TestSampleN(bshape, op)
+            if hybridize:
+                net.hybridize()
+            expected_shape = (shape_formatter(bshape) +
+                              shape_formatter(eshape))
+            out = net(np.ones(shape=eshape), np.ones(shape=eshape))
+            assert out.shape == expected_shape
```
Comment (sxjscience) [inline tests/python/unittest/test_numpy_op.py]: We need some randomness tests by calling https://github.com/apache/incubator-mxnet/blob/4da14a22385622c35e9a5c9c3e8a17c07f718cad/python/mxnet/test_utils.py#L2199-L2245

Comment (xidulu) [reply tests/python/unittest/test_numpy_op.py]: Running chi-square test will take too much time for a unittest,
I'll add one to the nightly test.

Comment (xidulu) [reply tests/python/unittest/test_numpy_op.py]: Tests implemented in nightly/test_np_random

File: tests/nightly/test_np_random.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+            buckets, probs = gen_buckets_probs_with_ppf(lambda x: ss.norm.pdf(x, loc=low, scale=scale), num_buckets)
+            buckets = np.array(buckets, dtype=dtype).tolist()
+            probs = [(buckets[i][1] - buckets[i][0])/scale for i in range(num_buckets)]
+            generator_mx_np = lambda x: np.random.normal(loc, scale, size=x, ctx=ctx, dtype=dtype).asnumpy()
+            verify_generator(generator=generator_mx_np, buckets=buckets, probs=probs, nsamples=samples, nrepeat=trials)
+
+
+if __name__ == '__main__':
+    import nose
+    nose.runmodule()
```
Comment (sxjscience) [inline tests/nightly/test_np_random.py]: Empty line

**Comment by sxjscience** [comment]
LGTM

File: tests/nightly/test_np_random.py (hunk 2 above)
Comment (xidulu) [reply tests/nightly/test_np_random.py]: fixed.

**Pull request merged**